### PR TITLE
add Pydantic model for Boa Recipe

### PR DESCRIPTION
**Pull request by madhur-tandon**
Solves https://github.com/mamba-org/boa/issues/209

One can regenerate the json by adding the statement: `print(BoaRecipeV1.schema_json(indent=2))`

### CODE REVIEW

**Comment by wolfv** [comment]
How can we go from this to the `json` file?

File: boa/schemas/model.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    skip: Optional[List[str]] = None
+    noarch: Optional[str] = None
+    pre_link: Optional[str] = Field(None, alias="pre-link")
+    post_link: Optional[str] = Field(None, alias="post-link")
+    pre_unlink: Optional[str] = Field(None, alias="pre-unlink")
+    ignore_run_exports: Optional[List[str]] = None
+    ignore_run_exports_from: Optional[List[str]] = None
+
+
+class Package1(BaseModel):
```
Comment (wolfv) [inline boa/schemas/model.py:44]: can we merge this with the regular Package?

File: boa/schemas/model.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, Extra, Field, conint
+
+
+class Package(BaseModel):
+    class Config:
+        extra = Extra.forbid
+
+    name: Optional[str] = Field(None, description="The package name")
```
Comment (wolfv) [inline boa/schemas/model.py:16]: I think this can be mandatory

File: boa/schemas/model.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    version: Optional[str] = Field(None, description="The package version")
```
Comment (wolfv) [inline boa/schemas/model.py:17]: as well as version should also be mandatory

Comment (madhur-tandon) [reply boa/schemas/model.py]: done

File: boa/schemas/model.py (hunk 2 above)
Comment (madhur-tandon) [reply boa/schemas/model.py]: done

File: boa/schemas/model.py (hunk 1 above)
Comment (madhur-tandon) [reply boa/schemas/model.py]: yes, we can. done

**Pull request merged**